en/services/component_def.md

```diff
@@ -0,0 +1,52 @@
+# Component Definition File
+
+A component information file is an XML file that describes the custom settings of a MAVLink component (like a gimbal or companion computer).
+
+The location of the file can be requested from a component.
+Usually the component hosts the file and distributes it using [MAVLink FTP](../services/ftp.md).
+
+The settings themselves can be queried and set using [MAVLink Parameters](../services/parameter.md) (a GCS client may use the file to generate a settings UI).
+
+> **Note** The component definition file mechanism is based on the [camera definition file](../services/camera_def.md), and therefore very similar.
+
+## Discovery Mechanism
+
+The ground station must first discover all components on the network by broadcasting the command [MAV_CMD_REQUEST_MESSAGE](../messages/common.md#MAV_CMD_REQUEST_MESSAGE) (specifying the [COMPONENT_INFORMATION](../messages/common.md#COMPONENT_INFORMATION) message).
+All components should respond with `COMPONENT_INFORMATION`.
+
+The `COMPONENT_INFORMATION.component_definition_uri` field contains the definition file URL.
+Typically this is a [MAVLink FTP URL](../services/ftp.md) address (i.e. the component itself), but it might also be an address on the Internet.
+
+## Example Definition File
```

Comment:
Can we do any standardisation on the XML format yet? If so we should collect and publish that.